core/chain_indexer.go

```diff
@@ -61,6 +64,8 @@ type ChainIndexerChain interface {
 	SubscribeChainHeadEvent(ch chan<- ChainHeadEvent) event.Subscription
 	// NodeCtx returns the context of the chain
 	NodeCtx() int
+	// StateAt returns the state for a root and utxo
```

Comment:
This comment doesn't makes sense. I think this method is supposed to return the Quai state trie for a state root, and the Qi UTXO set for a UTXO root, right?

Reply:
StateAt returns a StateDB which would contain both Quai and Qi tries. Updated the comment to be more precise.